packages/graph/hash_graph/tests/integration/postgres/mod.rs

```diff
@@ -0,0 +1,150 @@
+mod data_type;
+
+use error_stack::Result;
+use graph::{
+    datastore::{
+        DatabaseConnectionInfo, DatabaseType, Datastore, InsertionError, PostgresDatabase,
+        QueryError, UpdateError,
+    },
+    types::{schema::DataType, AccountId, BaseUri, Qualified, VersionId},
+};
+use sqlx::{Connection, PgConnection};
+use tokio::runtime::Runtime;
+use uuid::Uuid;
+
+pub struct Database {
+    postgres: PostgresDatabase,
+    connection: Option<PgConnection>,
+    created_base_uris: Vec<BaseUri>,
+    account_id: AccountId,
+    rt: Runtime,
+}
+
+impl Database {
+    pub fn new() -> Self {
+        const USER: &str = "postgres";
+        const PASSWORD: &str = "postgres";
+        const HOST: &str = "localhost";
+        const PORT: u16 = 5432;
+        const DATABASE: &str = "graph";
+
+        let connection_info = DatabaseConnectionInfo::new(
+            DatabaseType::Postgres,
+            USER.to_owned(),
+            PASSWORD.to_owned(),
+            HOST.to_owned(),
+            PORT,
+            DATABASE.to_owned(),
+        );
+
+        let rt = Runtime::new().expect("Could not create a test runtime");
+        let (postgres, connection, account_id) = rt.block_on(async {
+            let postgres = PostgresDatabase::new(&connection_info)
+                .await
+                .expect("could not connect to database");
+
+            let mut connection = PgConnection::connect(&connection_info.url())
+                .await
+                .expect("could not connect to database");
+
+            let account_id = create_account_id(&mut connection).await;
+
+            (postgres, connection, account_id)
+        });
+
+        Self {
+            postgres,
+            connection: Some(connection),
+            created_base_uris: Vec::new(),
+            account_id,
+            rt,
+        }
+    }
+
+    pub fn create_data_type(
+        &mut self,
+        data_type: DataType,
+    ) -> Result<Qualified<DataType>, InsertionError> {
+        self.rt.block_on(async {
+            let data_type = self
+                .postgres
+                .create_data_type(data_type, self.account_id)
```

Comment:
I'm fine with us leaving it for now, but I just to flag that I think we can't just abstract over the account_id for long as we're going to have to test with multiple accounts and such.

Reply:
As we currently have no logic for authentication, I provided the id from the wrapper. Once we have the logic in place this will be replaced.